forc-plugins/forc-client/src/util/tx.rs

```
@@ -165,9 +187,30 @@ impl<Tx: Buildable + SerializableVec + field::Witnesses + Send> TransactionBuild
                         "\nPlease provide the password of your encrypted wallet vault at {wallet_path:?}:"
                     );
                     let password = rpassword::prompt_password(prompt)?;
-                    // TODO: List all derived wallets via forc-wallet and let the users choose
-                    // account.
-                    let account_index = 0;
+                    let verification = AccountVerification::Yes(password.clone());
+                    let accounts = collect_accounts_with_verification(&wallet_path, verification)?;
+                    let provider = Provider::new(client.clone(), params);
+                    let account_balances = collect_account_balances(&accounts, &provider).await?;
+
+                    let total_balance = account_balances
+                        .iter()
+                        .flat_map(|account| account.values())
+                        .sum::<u64>();
+                    if total_balance == 0 {
```

Should we throw an error if the balance is somehow negative?